### DISCUSSION TITLE
How  to get thumbnail of iModel ?

### DISCUSSION THREAD

**Discussion by nawazmujawar1** [iTwin/itwinjs-core#2804] [Q&A]
I want to show the thumbnail of iModel as preview image

**Comment by aruniverse** [comment]
Afaik thumbnails aren't automatically generated, and you'll have to create those yourself. Here's an example of how to capture the viewport and get an exportable thumbnail: https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=image-export-sample&imodel=Metrostation+Sample

**Comment by JohnMajerle** [comment]
Nice sample.  Is it possible to export an image from an agent instead of a viewer?  We would like to export a potentially large collection of images which would likely be computationally expensive to do from an interactive viewer.

**Reply by pmconne** [reply]
Your agent would need to run the frontend to produce the images - WebGL is only available in a browser or electron.

**Comment by ramanujam-raman-zz** [comment]
There's API to get the thumbnails stored with the hub - please take a look  - 
https://www.itwinjs.org/reference/imodelhub-client/imodelhubclient/thumbnail/

Tests that use the above - https://github.com/iTwin/itwinjs-core/blob/release/2.19.x/full-stack-tests/imodelhub-client/src/imodelhub/Thumbnails.test.ts

**Reply by pmconne** [reply]
@ramanujam-raman someone has to produce and upload those thumbnails though, no?

**Reply by aruniverse** [reply]
Correct. Here are the rest apis for [upload](https://developer.bentley.com/apis/imodels/operations/upload-imodel-thumbnail/) and [download](https://developer.bentley.com/apis/imodels/operations/get-imodel-thumbnail/), but there seems to be only 1 thumbnail per model here.
